contracts/BaseProvider/BaseLockDealProvider.sol

```diff
@@ -23,7 +27,7 @@ contract BaseLockDealProvider is
         address token,
         uint256[] memory params
     ) public returns (uint256 poolId) {
-        poolId = lockDealNFT.mint(owner, token, params[0]);
+        poolId = lockDealNFT.mint(owner,  token, msg.sender, params[0]);
```

Comment:
I think its better to explicitly pass the address,
so the handle can open a pool for other provider

Reply:
Hmm, I don't quite get it. 
Pass the provider's address to the mint function?